File: example/src/main/java/com/vimeo/android/networking/example/TestApp.java
package com.vimeo.android.networking.example;

import android.app.Application;
import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Build;
import android.util.Log;

import com.vimeo.android.networking.example.vimeonetworking.AndroidGsonDeserializer;
import com.vimeo.android.networking.example.vimeonetworking.NetworkingLogger;
import com.vimeo.android.networking.example.vimeonetworking.TestAccountStore;
import com.vimeo.networking.Configuration;
import com.vimeo.networking.Vimeo.LogLevel;
import com.vimeo.networking.VimeoClient;

/**
 * Created by kylevenn on 1/27/16.
 */
public class TestApp extends Application {

    private static final String SCOPE = "private public create edit delete interact";

    private static final boolean IS_DEBUG_BUILD = false;
    // Switch to true to see how access token auth works.
    private static final boolean ACCESS_TOKEN_PROVIDED = false;

    @Override
    public void onCreate() {
        super.onCreate();

        AccountPreferenceManager.initializeInstance(this);

        // <editor-fold desc="Vimeo API Library Initialization">
        final Configuration.Builder configBuilder;
        // This check is just as for the example. In practice, you'd use one technique or the other.
        if (ACCESS_TOKEN_PROVIDED) {
            configBuilder = getAccessTokenBuilder();
        } else {
            configBuilder = getClientIdAndClientSecretBuilder();
        }
        if (IS_DEBUG_BUILD) {
            // Disable cert pinning if debugging (so we can intercept packets)
            configBuilder.enableCertPinning(false);
            configBuilder.setLogLevel(LogLevel.VERBOSE);
        }
        VimeoClient.initialize(configBuilder.build());
        // </editor-fold>
    }

    public Configuration.Builder getAccessTokenBuilder() {
        // The values file is left out of git, so you'll have to provide your own access token
        final String accessToken = getString(R.string.access_token);
        return new Configuration.Builder(accessToken);
    }

    public Configuration.Builder getClientIdAndClientSecretBuilder() {
        // The values file is left out of git, so you'll have to provide your own id and secret
        final String clientId = getString(R.string.client_id);
        final String clientSecret = getString(R.string.client_secret);
        final String codeGrantRedirectUri = getString(R.string.deeplink_redirect_scheme) + "://" +
                                            getString(R.string.deeplink_redirect_host);
        final TestAccountStore testAccountStore = new TestAccountStore(this.getApplicationContext());
        final Configuration.Builder configBuilder =
                new Configuration.Builder(clientId, clientSecret, SCOPE, testAccountStore,
                                          new AndroidGsonDeserializer());
        configBuilder.setCacheDirectory(this.getCacheDir())
                .setUserAgentString(getUserAgentString(this)).setDebugLogger(new NetworkingLogger())
                // Used for oauth flow
                .setCodeGrantRedirectUri(codeGrantRedirectUri);

        return configBuilder;
    }

    public static String getUserAgentString(Context context) {
        final String packageName = context.getPackageName();

        String version = "unknown";
        try {
            final PackageInfo pInfo = context.getPackageManager().getPackageInfo(packageName, 0);
            version = pInfo.versionName;
        } catch (final PackageManager.NameNotFoundException e) {
            Log.e("TestApp","Unable to get packageInfo: " + e.getMessage());
        }

        final String deviceManufacturer = Build.MANUFACTURER;
        final String deviceModel = Build.MODEL;
        final String deviceBrand = Build.BRAND;

        final String versionString = Build.VERSION.RELEASE;
        final String versionSDKString = String.valueOf(Build.VERSION.SDK_INT);

        return packageName + " (" + deviceManufacturer + ", " + deviceModel + ", " + deviceBrand +
               ", " + "Android " + versionString + "/" + versionSDKString + " Version " + version +
               ")";
    }
}

File: example-kotlin-android/src/main/kotlin/com/vimeo/android/networking/example/kotlin/TestApp.kt
package com.vimeo.android.networking.example.kotlin

import android.app.Application
import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import com.vimeo.android.networking.example.kotlin.vimeonetworking.NetworkingLogger
import com.vimeo.networking.Configuration
import com.vimeo.networking.Vimeo.LogLevel
import com.vimeo.networking.VimeoClient

/**
 * The application class.
 */
@Suppress("ConstantConditionIf")
class TestApp : Application() {

    override fun onCreate() {
        super.onCreate()
        AccountPreferenceManager.initializeInstance(this)

        // This check is just as for the example. In practice, you'd use one technique or the other.
        val configBuilder =
                if (ACCESS_TOKEN_PROVIDED) accessTokenBuilder
                else clientIdAndClientSecretBuilder

        configBuilder.apply {
            // Disable cert pinning if debugging (so we can intercept packets)
            if (BuildConfig.DEBUG) {
                enableCertPinning(false)
                setLogLevel(LogLevel.VERBOSE)
            }

            setCacheDirectory(cacheDir)
            setUserAgentString(getUserAgentString(this@TestApp))
            setDebugLogger(NetworkingLogger())
        }

        VimeoClient.initialize(configBuilder.build())
    }

    // The values file is left out of git, so you'll have to provide your own access token
    private val accessTokenBuilder: Configuration.Builder
        get() {
            val accessToken = ""
            return Configuration.Builder(accessToken)
        }

    // The values file is left out of git, so you'll have to provide your own id and secret
    // Used for oauth flow
    private val clientIdAndClientSecretBuilder: Configuration.Builder
        get() {
            val clientId = "PROVIDE A CLIENT ID"
            val clientSecret = "PROVIDE A CLIENT SECRET"
            val codeGrantRedirectUri = getString(R.string.deeplink_redirect_scheme) + "://" + getString(R.string.deeplink_redirect_host)

            return Configuration.Builder(clientId, clientSecret, SCOPE).apply {
                setCodeGrantRedirectUri(codeGrantRedirectUri)
            }
        }

    companion object {

        private const val SCOPE = "private public create edit delete interact"

        // Switch to true to see how access token auth works.
        private const val ACCESS_TOKEN_PROVIDED = false

        fun getUserAgentString(context: Context): String {
            val packageName = context.packageName

            var version = "unknown"
            try {
                val pInfo = context.packageManager.getPackageInfo(packageName, 0)
                version = pInfo.versionName
            } catch (e: PackageManager.NameNotFoundException) {
                println("Unable to get packageInfo: " + e.message)
            }

            return "$packageName (${Build.MANUFACTURER}, ${Build.MODEL}, ${Build.BRAND}, Android ${Build.VERSION.RELEASE}/${Build.VERSION.SDK_INT} Version $version)"
        }
    }
}
